src/main/java/org/ibp/api/java/impl/middleware/study/StudyServiceImpl.java

```diff
@@ -381,4 +382,26 @@ public void updateStudy(final Study study) {
 		this.studyValidator.validate(studyId, false);
 		this.studyDataManager.updateStudyLockedStatus(studyId, study.isLocked());
 	}
+
+	@Override
+	public long countVariablesByDatasetId(final int studyDbId, final List<Integer> variableTypes) {
```

Comment:
The parameter is `studyDbId `but the Middleware call is countVariablesBy**DatasetId**?